crates/kornia-tracking/src/image_utilities.rs

```diff
@@ -0,0 +1,146 @@
+//! Image utilities used by the kornia-tracking modules.
+//! This module contains functions to compute image gradients, check bounds,
+//! perform the SE(2) exponential map, and detect keypoints.
+
+use kornia_image::Image;
+use glam::{Mat3, Vec3};
+use kornia_imgproc::corners::{corners_fast9,Corner};
+
+/// Computes image gradients at (x, y).
+pub fn image_grad(grayscale_image: &Image<f32, 1>, x: f32, y: f32) -> Vec3 {
```

Comment:
```suggestion
pub fn image_grad(grayscale_image: &Image<f32, 1>, x: f32, y: f32) -> [f32; 3] {
```
this should go to kornia-imgproc